packages/ui5-test-writer/src/index.ts

```diff
@@ -0,0 +1,209 @@
+ * @returns OPA test configuration object
+ */
+function createConfig(manifest: Manifest, opaConfig: { scriptName?: string }): FEV4OPAConfig {
+    // General application info
+    const { appID, appPath } = getAppFromManifest(manifest);
+
+    const config: FEV4OPAConfig = {
+        appID,
+        appPath,
+        pages: [],
+        opaJourneyFileName: opaConfig.scriptName || 'FirstJourney'
+    };
+
+    // Identify startup targets from the routes
+    const appRoutes = (manifest['sap.ui5']?.routing?.routes ?? []) as any[];
+    // Find the route with an empty pattern (except for the trailing query part)
+    const startupRoute = appRoutes.find((route: { pattern: string }) => {
+        return route.pattern.replace(':?query:', '') === '';
+    });
+    let startupTargets = startupRoute?.target ?? [];
+    if (!Array.isArray(startupTargets)) {
+        startupTargets = [startupTargets];
+    }
+
+    // Create page configurations in supported cases
+    const appTargets = manifest['sap.ui5']?.routing?.targets;
+    for (const targetKey in appTargets) {
+        const pageConfig = createPageConfig(manifest, targetKey);
+        if (pageConfig) {
+            pageConfig.isStartup = startupTargets.includes(targetKey);
+            config.pages.push(pageConfig);
+        }
+    }
+
+    return config;
+}
+
+/**
+ * Writes a page object in a mem-fs-editor.
+ *
+ * @param pageConfig - the page configuration object
+ * @param rootTemplateDirPath - template root directory
+ * @param testOutDirPath - output test directory (.../webapp/test)
+ * @param fs - a reference to a mem-fs editor
+ */
+function writePageObject(
+    pageConfig: FEV4OPAPageConfig,
+    rootTemplateDirPath: string,
+    testOutDirPath: string,
+    fs: Editor
+) {
+    fs.copyTpl(
+        join(rootTemplateDirPath, `integration/pages/${pageConfig.template}.js`),
+        join(testOutDirPath, `integration/pages/${pageConfig.targetKey}.js`),
+        pageConfig,
+        undefined,
+        {
+            globOptions: { dot: true }
+        }
+    );
+}
+
+/**
+ * Generate OPA test files for a Fiori elements for OData V4 application.
+ * Note: this can potentially overwrite existing files in the webapp/test folder.
+ *
+ * @param basePath - the absolute target path where the application will be generated
+ * @param opaConfig - parameters for the generation
+ * @param opaConfig.scriptName - the name of the OPA journey file. If not specified, 'FirstJourney' will be used
+ * @param fs - an optional reference to a mem-fs editor
+ * @returns Reference to a mem-fs-editor
+ */
+export function generateOPAFiles(basePath: string, opaConfig: { scriptName?: string }, fs?: Editor): Editor {
+    if (!fs) {
+        fs = create(createStorage());
+    }
+
+    const manifest = fs.readJSON(join(basePath, 'webapp/manifest.json')) as any as Manifest;
+    const config = createConfig(manifest, opaConfig);
```

Comment:
Do we validate to check if the app is of an unsupported version/floor plan?

Reply:
I don't know how to check that...

Reply:
Can we leave it to the consumer to set the required version through API parameters so we do not concern the generation of the tests with validation?